tests/components/vesync/test_init.py

```diff
@@ -0,0 +1,198 @@
+"""Tests for the init module."""
+from unittest.mock import Mock, patch
+
+from pyvesync import VeSync
+
+from homeassistant.components.vesync import async_setup_entry
+from homeassistant.components.vesync.const import (
+    DOMAIN,
+    VS_FANS,
+    VS_LIGHTS,
+    VS_MANAGER,
+    VS_SENSORS,
+    VS_SWITCHES,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.setup import async_setup_component
+
+
+async def test_async_setup_component(hass: HomeAssistant, config: ConfigType):
+    """Test setup component."""
+    assert await async_setup_component(hass, DOMAIN, config)
+    await hass.async_block_till_done()
+
+
+async def test_async_setup_entry__not_login(
+    hass: HomeAssistant, config_entry: ConfigEntry, manager: VeSync, caplog
+):
+    """Test setup does not create config entry when not logged in."""
+    manager.login = Mock(return_value=False)
+
+    assert not await async_setup_entry(hass, config_entry)
```

Comment:
After this call, verify the state of the config entry is as expected.

Reply:
I'm not sure what I should verify on config_entry because there are no changes to it...unless that is what I need to verify?  I added code to verify that the setup does not proceed if not logged in.